core-rust/state-manager/src/transaction/preview.rs

```diff
@@ -68,24 +62,21 @@ impl<S: ReadableStore + QueryableProofStore + TransactionIdentifierLoader> Trans
             .execution_configurator
             .wrap(validated.get_executable(), ConfigType::Preview)
             .warn_after(PREVIEW_RUNTIME_WARN_THRESHOLD, "preview");
-        let receipt = transaction_logic.execute_on(read_store.deref());
 
-        // Fake a LedgerPayloadHash for the purposes of mapping the receipt as it doesn't matter for preview
-        // TODO - don't do most of this work for preview
```

Comment:
_(note from author)_

I decided to address this now, since otherwise I would have to fake even more stuff.